include/Note.h

```diff
@@ -247,6 +261,9 @@ class LMMS_EXPORT Note : public SerializingObject
 	TimePos m_length;
 	TimePos m_pos;
 	DetuningHelper * m_detuning;
+
+	// The type of this note
+	Type m_type;
```

Comment:
I would suggest to also use an in-class initializer here.
```suggestion
	Type m_type = Types::RegularNote;
```